pandas/core/frame.py

```diff
@@ -553,17 +553,22 @@ def __init__(
         index: Optional[Axes] = None,
         columns: Optional[Axes] = None,
         dtype: Optional[Dtype] = None,
-        copy: bool = False,
+        copy: Optional[bool] = None,
     ):
+        orig_copy = copy  # GH#38939
+        copy = copy if copy is not None else False
```

Comment:
`copy = copy if copy and isinstance(data, dict) else False` ? (and remove orig_copy)

Reply:
we also redefine `copy` in two more places below.  Id have to go back and ask past-me, but Im pretty sure that wasnt his first choice, so there must have been a reason